qiskit/algorithms/time_evolvers/variational/solvers/var_qte_linear_solver.py

```diff
@@ -0,0 +1,116 @@
+            hamiltonian: TODO update types
+                Operator used for Variational Quantum Time Evolution.
+                The operator may be given either as a composed op consisting of a Hermitian
+                observable and a ``CircuitStateFn`` or a ``ListOp`` of a ``CircuitStateFn`` with a
+                ``ComboFn``.
+                The latter case enables the evaluation of a Quantum Natural Gradient.
+            ansatz: Quantum state in the form of a parametrized quantum circuit.
+            gradient_params: List of parameters with respect to which gradients should be computed.
+            t_param: Time parameter in case of a time-dependent Hamiltonian.
+            lse_solver: Linear system of equations solver callable. It accepts ``A`` and ``b`` to
+                solve ``Ax=b`` and returns ``x``. If ``None``, the default ``np.linalg.lstsq``
+                solver is used.
+            imag_part_tol: Allowed value of an imaginary part that can be neglected if no
+                imaginary part is expected.
+        """
+        self._var_principle = var_principle
+        self._hamiltonian = hamiltonian
+        self._ansatz = ansatz
+        self._gradient_params = gradient_params
+        self._bind_params = gradient_params
+        self._time_param = t_param
+        self.lse_solver = lse_solver
+        self._imag_part_tol = imag_part_tol
+
+    @property
+    def lse_solver(self) -> Callable[[np.ndarray, np.ndarray], np.ndarray]:
+        """Returns an LSE solver callable."""
+        return self._lse_solver
+
+    @lse_solver.setter
+    def lse_solver(self, lse_solver: Callable[[np.ndarray, np.ndarray], np.ndarray] | None) -> None:
+        """Sets an LSE solver. Uses a ``np.linalg.lstsq`` callable if ``None`` provided."""
+        if lse_solver is None:
+            lse_solver = lambda a, b: np.linalg.lstsq(a, b, rcond=1e-2)[0]
+
+        self._lse_solver = lse_solver
+
+    def solve_lse(
+        self,
+        param_dict: dict[Parameter, complex],
+        time_value: float | None = None,
+    ) -> (list | np.ndarray, list | np.ndarray, np.ndarray):
+        """
+        Solve the system of linear equations underlying McLachlan's variational principle for the
+        calculation without error bounds.
+
+        Args:
+            param_dict: Dictionary which relates parameter values to the parameters in the ansatz.
+            time_value: Time value that will be bound to ``t_param``. It is required if ``t_param``
+                is not ``None``.
+
+        Returns:
+            Solution to the LSE, A from Ax=b, b from Ax=b.
+        """
+        param_values = list(param_dict.values())
+        metric_tensor_lse_lhs = self._var_principle.metric_tensor(self._ansatz, param_values)
+        if self._time_param is not None:
+            # TODO this turns into OperatorBase
+            self._hamiltonian = self._hamiltonian.assign_parameters({self._time_param: time_value})
```

Comment:
It seems like this would not be re-usable if we do the substitution in place, right?

Reply:
Good catch, converting to a local variable.